pkg/agent/nodeportlocal/k8s/controller.go

```diff
@@ -64,28 +117,31 @@ func podKeyFunc(pod *corev1.Pod) string {
 
 // Run starts to watch and process Pod updates for the Node where Antrea Agent is running.
 // It starts a queue and a fixed number of workers to process the objects from the queue.
-func (c *Controller) Run(stopCh <-chan struct{}) {
+func (c *NPLController) Run(stopCh <-chan struct{}) {
 	defer c.queue.ShutDown()
 	for i := 0; i < numWorkers; i++ {
 		go wait.Until(c.Worker, time.Second, stopCh)
 	}
-	c.Ctrl.Run(stopCh)
-	cache.WaitForCacheSync(stopCh, c.Ctrl.HasSynced)
+
+	go c.podInformer.Run(stopCh)
+	go c.svcInformer.Informer().Run(stopCh)
```

Comment:
I think `informerFactory.Start(stopCh)` will start this. There may be issue to run it twice.